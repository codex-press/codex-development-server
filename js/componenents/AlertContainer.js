import React from 'react';
import { connect } from 'react-redux'
import Alert from './Alert';

export default function AlertContainer(props) {

  return (
    <div className="AlertContainer">
      { props.alerts && props.alerts.toArray().map(a =>
        <Alert {...a} key={ a.id } onClick={ () => props.remove(a.id) } />)
      }
    </div>
  );
}




