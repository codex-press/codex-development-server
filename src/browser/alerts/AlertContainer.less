@import (reference) "../index.less";

.AlertContainer {
  position: fixed;
  top: 2rem;
  left: 0;
  right: 0;
  margin: auto;
  width: 500px;
  max-width: calc(100% - 2rem);
  z-index: (@maxZIndex - 1);
}

