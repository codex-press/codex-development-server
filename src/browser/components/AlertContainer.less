@import (reference) "../index.less";

.AlertContainer {
  position: fixed;
  top: 3em;
  left: 50%;
  transform: translateX(-50%);
  width: 90%;
  max-width: 600px;
  z-index: 100000;
  line-height: 1.2;
  font-family: "Helvetica Neue", Helvetica, Arial, sans-serif;
  color: #333;
  z-index: (@maxZIndex - 1);
}

