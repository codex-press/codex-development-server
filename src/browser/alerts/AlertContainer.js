import React from 'react';
import { connect } from 'react-redux'

import { removeAlert } from '../actions';
import Alert from './Alert';

import './AlertContainer.less';


const mapStateToProps = state => {
  return {
    alerts: state.get('alerts'),
  }
}


const mapDispatchToProps = {
  removeAlert,
}



export function AlertContainer(props) {

  return (
    <div className="AlertContainer">
      { props.alerts.toArray().map(a =>
        <Alert
          {...a.toJS()}
          key={ a.get('id') }
          remove={ () => props.removeAlert(a.get('id')) }
        />)
      }
    </div>
  );
}


export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(AlertContainer);


