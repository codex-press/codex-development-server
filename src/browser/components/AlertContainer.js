import React from 'react';
import { connect } from 'react-redux'

import { removeAlert } from '../actions';
import Alert from './Alert';

import './AlertContainer.less';


const mapStateToProps = state => {
  return {
    alerts: state.get('alerts'),
  }
}


const mapDispatchToProps = {
  removeAlert,
}



export function AlertContainer(props) {

  return (
    <div className="AlertContainer">
      { props.alerts.toArray().map(a =>
        <Alert
          {...a}
          key={ a.id }
          remove={ () => props.removeAlert(a.id) }
        />)
      }
    </div>
  );
}


export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(AlertContainer);


