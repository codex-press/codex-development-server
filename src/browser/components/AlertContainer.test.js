import React from 'react';
import { shallow } from 'enzyme';
import { Map } from 'immutable';

import { AlertContainer } from './AlertContainer';

test('Renders an alert', () => {
  const alerts = Map({
    connection: { id: 'connection', head: 'Lost Connection' }
  });
  expect(shallow(
    <AlertContainer alerts={ alerts } />
  )).toMatchSnapshot();
});


