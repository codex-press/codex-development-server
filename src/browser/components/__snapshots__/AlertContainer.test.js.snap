// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders an alert 1`] = `
<div
  className="AlertContainer"
>
  <Alert
    head="Lost Connection"
    id="connection"
    remove={[Function]}
  />
</div>
`;
